### DISCUSSION TITLE
how to hide the env value which  can be seen by any one in browser's sources tab

### DISCUSSION THREAD

**Discussion by vaibhavk2474** [vercel/next.js#64185] [Help]
### Summary

i am using next 13 and i want to hide env value from sources tab where it is clearly visible there is any way to hide it because source map is  also not disabling after writing   productionBrowserSourceMaps: false, in next config file

**Comment by icyJoseph** [comment]
How come it ends up in the source, is it a NEXT_PUBLIC kind of value? if its only used server side, then remove the NEXT_PUBLIC prefix.

**Reply by vaibhavk2474** [reply]
actually i am using that value in client side code i need to used that NEXT_PUBLIC prefix but is there  any possible way, not to show these values on bowser or disable source map?
 because i tried productionBrowserSourceMaps:false  also, but it is  not working

**Reply by icyJoseph** [reply]
If it is included in the javascript/html or css that runs that site, it is public to anyone. Best you can do is obfuscate it, or change your logic so that the value is fetched from the server rather, but that'd introduce a bit of asynchronous work to your app.

**Reply by vaibhavk2474** [reply]
you mean i should get that secret values from server but in network it can be visible then i need to use encryption also. 
am i right?

**Reply by icyJoseph** [reply]
I am not sure you are on the right track with this.

If you send it from the server, encrypted, it will have to decrypted client side, which means it is visible client side. One could just add a debugger breakpoint, and step through to see the value. 

If it will arrive to the client, then it will be visible. You can only make it more difficult.

**Comment by evansgreatest** [comment]
In your project root, create a .env.local file (or any other .env.* file) and define your environment variables there.

**Reply by vaibhavk2474** [reply]
I am doing that but when I used NEXT_PUBLIC_ 
 It can be visible to browser source tab and if I don't use it then I can not use it in client side also

**Reply by evansgreatest** [reply]
Define env vars in .env.local, don't use NEXT_PUBLIC_
eg. MY_KEY=your_key_value
